pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -304,7 +309,11 @@ func ingestWithRetries(
 		log.Warningf(ctx, msgFmt, err)
 		updateRunningStatus(ctx, ingestionJob, jobspb.ReplicationError,
 			fmt.Sprintf(msgFmt, err))
-		retryCount++
+		latestReplicatedTime := loadRetainedTime(ctx, execCtx.ExecCfg().InternalDB, ingestionJob)
+		if lastReplicatedTime.Less(latestReplicatedTime) {
```

Comment:
Should this be LessEq?

Reply:
Why do you think that? Perhaps my naming was confusing.

We should only reset the retry counter if the replicated time has advanced, i.e `lastReplicatedTime.Less(newReplicatedTime)`

Reply:
Yeah, I just misread. You are good to go.